action.yml

```diff
@@ -23,38 +21,56 @@ runs:
       name: Label Artifact Version
       uses: docker://esolang/jq:latest
       run: |
-        #Create payload
-        echo '{ "component_id": "'"$INPUT_COMPONENT_ID"'", "artifact_id": "'"$INPUT_ARTIFACT_ID"'", ' > /tmp/payload.json 
+        # Set default operation to ADD if not provided
+        OPERATION=${INPUT_OPERATION:-ADD}
+
+        # Create payload (without operation)
+        echo '{ "artifact_id": "'"$INPUT_ARTIFACT_ID"'", ' > /tmp/payload.json
+        formatted_labels=$(echo "$INPUT_LABELS" | sed -e 's/[][]//g' | awk -v RS=, -v ORS=, '{ gsub(/^[ \t\r\n]+|[ \t\r\n]+$/, "");  gsub(/\\"/, "\\\\\""); gsub(/"/, "\\\""); printf "\"%s\",", $0 }' | sed 's/,$//' | sed 's/^/[/; s/$/]/')
```

Comment:
this is a pretty hard to read command. Can we add comments to explain the expected format for the input that this is trying to parse?